subsys/power/pm_ctrl.c

```diff
@@ -14,15 +14,15 @@
 #include <logging/log.h>
 LOG_MODULE_DECLARE(power);
 
-static atomic_t power_state_disable_count[SYS_POWER_STATE_MAX];
+static atomic_t power_state_enable_count[SYS_POWER_STATE_MAX];
```

Comment:
The functionality of the reworked code is meant to be identical with the only difference that the power states are disabled by default. Previous code was actually easier to understand, however would require adding `SYS_INIT` macro to call a routine that would initialize `power_state_disable_count` array with ones. Also an option.